### PR TITLE
Docker multi stage build

### PR DESCRIPTION
**Pull request by Hemankita**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,8 +1,25 @@
...
 
-COPY ./build/libs/*.jar ./server.jar
+COPY --from=builder /home/gradle/src/build/libs/*.jar ./server.jar/
+# Create app directory
+ENV APP_HOME=/app
+WORKDIR $APP_HOME
+
+# Copy jar file over from builder stage
+COPY --from=builder /home/gradle/src/build/libs/template-java-spring-1.0.0.jar $APP_HOME
+RUN mv ./template-java-spring-1.0.0.jar server.jar
```
Comment (csantanapr) [inline Dockerfile]: I get errors with this Dockerfile
```
 ---> f3263e864eb7
Step 9/15 : COPY --from=builder /home/gradle/src/build/libs/*.jar ./server.jar/
 ---> 94f7ca2ad506
Step 10/15 : ENV APP_HOME=/app
 ---> Running in c22a51311625
Removing intermediate container c22a51311625
 ---> 3739adf709a6
Step 11/15 : WORKDIR $APP_HOME
 ---> Running in 90cf41f8d0e2
Removing intermediate container 90cf41f8d0e2
 ---> 22c29173fefd
Step 12/15 : COPY --from=builder /home/gradle/src/build/libs/template-java-spring-1.0.0.jar $APP_HOME
COPY failed: stat /var/lib/docker/overlay2/a44087818f45ae900b9396c52632a438f1170b800dda809a54afb87e62f1b816/merged/home/gradle/src/build/libs/template-java-spring-1.0.0.jar: no such file or directory
```

File: Dockerfile [hunk 2]
```diff
@@ -1,8 +1,25 @@
...
 RUN dnf install -y java-11-openjdk.x86_64
 
-COPY ./build/libs/*.jar ./server.jar
+COPY --from=builder /home/gradle/src/build/libs/*.jar ./server.jar/
+# Create app directory
+ENV APP_HOME=/app
+WORKDIR $APP_HOME
+
+# Copy jar file over from builder stage
+COPY --from=builder /home/gradle/src/build/libs/template-java-spring-1.0.0.jar $APP_HOME
```
Comment (csantanapr) [inline Dockerfile]: Why there are 2 copies from the builder layer?

**Comment by csantanapr** [comment]
@Hemankita this Dockerfile works for me

```Dockerfile
FROM gradle:jdk11 AS builder

WORKDIR /home/gradle
COPY . .
RUN ./gradlew assemble --no-daemon  

FROM registry.access.redhat.com/ubi8/ubi

RUN dnf install -y java-11-openjdk.x86_64

COPY --from=builder /home/gradle/build/libs/*.jar ./server.jar

EXPOSE 9080/tcp

CMD ["java", "-jar", "./server.jar"]
```

**Comment by csantanapr** [comment]
found another problem while testing with tekton pipeline, that app name in package.json needs to match chart directory if not lint fails

**Comment by mjperrins** [comment]
We have identified from SFO workshop that pair should match and adding some extra checking for this, we able to check if the Dockfile builds OK for IKS install and with Jenkins ? @seansund was adding in Buildah and we test tomorrow

**Pull request closed without merging**